server/embed/config.go

```diff
@@ -1116,14 +1116,6 @@ func (cfg *Config) getListenClientUrls() (ss []string) {
 	return ss
 }
 
-func (cfg *Config) getListenClientHttpUrls() (ss []string) {
-	ss = make([]string, len(cfg.ListenClientHttpUrls))
-	for i := range cfg.ListenClientHttpUrls {
-		ss[i] = cfg.ListenClientHttpUrls[i].String()
-	}
-	return ss
-}
-
```

Comment:
Should we (1) remove all the `getXXXXUrls` methods or (2) exports all such methods and un-export the related fields? It's a little weird that the fields (e.g. `ListenClientUrls`, `ListenPeerUrls`,  `AdvertiseClientUrls`, `AdvertisePeerUrls` and `ListenClientHttpUrls`)  are exported, but the related methods are not. It can be discussed separately.

Reply:
I think we can keep and export the `getXXXXUrls` to make the copy of the fields, since etcd codebase prefers to use copy of source, for instance, pb use arg instead of pointer. If we un-exports the related fields, we need to provide the modifier option during configuration. It's refactor. We can do it in follow-up if it's accepted.